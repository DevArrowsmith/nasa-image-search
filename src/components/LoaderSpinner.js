import React from 'react';
import styled from 'styled-components';
import Loader from 'react-loader-spinner'

import "react-loader-spinner/dist/loader/css/react-spinner-loader.css"

const LoaderContainer = styled.div`
  height: 90vh;
  width: 100vw;
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
`

const LoaderSpinner = ({ resultsLoading }) => {
  if (resultsLoading) {
    return (
      <LoaderContainer>
        <Loader
          type="RevolvingDot"
          color="red"
          height={100}
          width={100}
          data-testid="Loader"
        />
      </LoaderContainer>
    );
  };
  return (null);
}

export default LoaderSpinner;
