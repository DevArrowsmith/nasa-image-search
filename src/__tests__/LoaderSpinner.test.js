import React from 'react';
import { render } from '@testing-library/react';
import LoaderSpinner from '../components/LoaderSpinner';

test('component LoaderSpinner renders', () => {
  const { asFragment } = render(<LoaderSpinner resultsLoading={true} />);
  const component = asFragment();
  expect(component).toMatchSnapshot();
});

test('component LoaderSpinner renders the Loading icon when resultsLoading state is true', () => {
  const { getByTestId } = render(<LoaderSpinner resultsLoading={true} />);
  const loader = getByTestId("Loader");
  expect(loader).toBeInTheDocument();
});

