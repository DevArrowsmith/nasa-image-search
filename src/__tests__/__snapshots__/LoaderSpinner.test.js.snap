// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component LoaderSpinner renders 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm doHGtG"
    data-testid="Loader"
  >
    <div
      aria-busy="true"
      class=""
    >
      <svg
        aria-label="audio-loading"
        height="100"
        version="1.1"
        width="100"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <circle
          cx="50"
          cy="50"
          fill="none"
          r="44"
          stroke="red"
          stroke-width="4"
          style="opacity: 0.5;"
        />
        <circle
          cx="8"
          cy="54"
          fill="red"
          r="6"
          stroke="red"
          stroke-width="3"
        >
          <animatetransform
            attributeName="transform"
            dur="2s"
            from="0 50 48"
            repeatCount="indefinite"
            to="360 50 52"
            type="rotate"
          />
        </circle>
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
